### PR TITLE
develop-book-category-tests-merge-to-master

### PR DESCRIPTION
**Pull request by alwayswannajava**
# Test's coverage
image

### CODE REVIEW

File: src/test/java/com/spring/onlinebookstore/service/book/CategoryServiceTest.java [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+        when(categoryMapper.toDto(category)).thenReturn(updateCategoryDto);
+        when(categoryRepository.findById(CORRECT_CATEGORY_ID))
+                .thenReturn(Optional.ofNullable(category));
+        when(categoryRepository.save(category)).thenReturn(category);
+
+        CategoryDto expected = updateCategoryDto;
+        CategoryDto actual = categoryService.update(CORRECT_CATEGORY_ID, updateCategoryRequestDto);
+
+        assertEquals(expected, actual);
+        verify(categoryRepository, times(1)).findById(CORRECT_CATEGORY_ID);
```
Comment (ahoienko) [inline src/test/java/com/spring/onlinebookstore/service/book/CategoryServiceTest.java]: Times 1 is set by default

File: src/test/resources/database/scripts/book/remove-book-test-data.sql [hunk 2]
```diff
@@ -0,0 +1,8 @@
+delete from books_categories;
+delete from categories;
+delete from books;
+
```
Comment (ahoienko) [inline src/test/resources/database/scripts/book/remove-book-test-data.sql]: Please remove redundant empty lines (check other scripts as well)

File: src/test/java/com/spring/onlinebookstore/service/book/CategoryServiceTest.java [hunk 3]
```diff
@@ -0,0 +1,161 @@
...
+    }
+
+    @Test
+    @DisplayName("Test update() non-existent book")
+    void update_NonExistentCategory_ShouldThrowEntityNotFoundException() {
+        when(categoryRepository.findById(INCORRECT_CATEGORY_ID)).thenReturn(Optional.empty());
+        assertThrows(EntityNotFoundException.class, () ->
```
Comment (ahoienko) [inline src/test/java/com/spring/onlinebookstore/service/book/CategoryServiceTest.java]: Please check error message as well

**Review by ahoienko — CHANGES_REQUESTED** [review]
Nice, left minor comments

File: src/test/java/com/spring/onlinebookstore/Constants.java [hunk 4]
```diff
@@ -0,0 +1,54 @@
+package com.spring.onlinebookstore;
+
+import com.spring.onlinebookstore.model.Category;
+import java.math.BigDecimal;
+import java.util.List;
+
+public final class Constants {
```
Comment (Elena-Bruyako) [inline src/test/java/com/spring/onlinebookstore/Constants.java]: it's better to replace to util package

File: src/test/java/com/spring/onlinebookstore/controller/BookControllerTest.java [hunk 5]
```diff
@@ -0,0 +1,271 @@
...
+        List<BookDto> actual = objectMapper.readValue(
+                result.getResponse().getContentAsString(),
+                new TypeReference<>() {}
+        );
+
+        EqualsBuilder.reflectionEquals(expected, actual, "id", "title",
+                "author", "price", "isbn", "categoryIds");
+    }
+
+    private void initDto() {
```
Comment (Elena-Bruyako) [inline src/test/java/com/spring/onlinebookstore/controller/BookControllerTest.java]: create TestUtil class and generate all test data there by using static methods
for all tests

**Review by Elena-Bruyako — CHANGES_REQUESTED** [review]
some comments

**Review by okuzan — APPROVED** [review]

**Pull request merged**